core/arch/arm/mm/mobj_dyn_shm.c

```diff
@@ -183,28 +184,41 @@ static TEE_Result mobj_reg_shm_inc_map(struct mobj *mobj)
 	uint32_t exceptions = 0;
 	size_t sz = 0;
 
-	if (refcount_inc(&r->mapcount))
-		return TEE_SUCCESS;
+	while (true) {
+		if (refcount_inc(&r->mapcount))
+			return TEE_SUCCESS;
```

Comment:
"When val is 0, refcount_inc() does not change the value and returns false."
So when `r->mapcount` is initially 0, how will any thread proceed?

Reply:
If val is 0 then we take the spinlock below and check that it's still 0 now that we have a lock. Then we map the mobj and initialize the mapcount to 1.

In case the val wasn't 0 any longer when we have the lock we instead increase the mapcount and return.

Reply:
Oh I'm sorry I totally missed the opening brace... :facepalm: thanks for clarifying

Reply:
No worries